File: Aula Seis -padrões de projeto singleton/singleton01.py
#Fatec Luigi Papaiz
#DSM - Desenvolvimento de Software Multiplataforma
#Disciplina - Técnicas de Programação II
#Nome: Thiago Messias Santos  22/03/2024
#Descrição: Aplicação do padrão GOF - Criacional - singleton

#codigo do livro

class Singleton:
    instance = None

    def __new__(cls):
        if cls.instance is None:
            cls.instance = super().__new__(cls)
            cls.instance.value = 0
        return cls.instance
    
    def increment(self):
        self.value += 1
        print(f"Value: {self.value}")


#cliente
        
s1 = Singleton()
s2 = Singleton()

print(s1 == s2)

s1.increment()
s2.increment()

File: aula 6 -padrões de projeto singleton/singleton01.js
//Fatec Luigi Papaiz
//DSM - Desenvolvimento de Software Multiplataforma
//Disciplina - Técnicas de Programação II
//Nome: Thiago Messias Santos  22/03/2024
//Descrição: Aplicação do padrão GOF - Criacional - singleton

//codigo do livro

class Singleton{

    constructor(){
        if(Singleton.instance == null){
            Singleton.instance = this;
            this.value = 0;
        }
        return Singleton.instance;
    }

    increment(){
        this.value += 1;
        console.log(`Value: ${this.value}`);

    }

}


//cliente:

const s1 = new Singleton();
const s2 = new Singleton();

console.log(s1 === s2);

s1.increment();
s2.increment();


